lib/to-interpolated-regexp.js

```diff
@@ -3,7 +3,7 @@ module.exports = function(source) {
     var splitSource = source.split('{componentName}');
     return function(componentName) {
       try {
-        return new RegExp(splitSource.length > 1 ? splitSource[0] + componentName + splitSource[1] : splitSource[0]);
+        return new RegExp(splitSource.join(componentName));
```

Comment:
Hmm, sure if you really need to see a failing test here, I can do it.

But to be honest, just check the docs of [`string.prototpe.split()`](String.prototype.split()) and [`Array.prototype.join()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join)

By splitting the `source` variable above, you get an Array of possible more than one items.
Hence you can just manually concatenate. Especially the check `splitSource.length > 1` here is very error prone.